components/select_team/components/select_team_item.jsx

```diff
@@ -59,14 +59,14 @@ export default class SelectTeamItem extends React.PureComponent {
         return (
             <div className='signup-team-dir'>
                 {showDescriptionTooltip}
-                <Link
-                    to={`/${this.props.team.name}/channels/town-square`}
```

Comment:
This was recently introduced and was the error. Routing is handled in parent, if the join is successful